src/language-js/printer-estree.js

```diff
@@ -4449,7 +4450,20 @@ function printMemberChain(path, options, print) {
     groups.length >= 2 && !groups[1][0].node.comments && shouldNotWrap(groups);
 
   function printGroup(printedGroup) {
-    return concat(printedGroup.map(tuple => tuple.printed));
+    const result = [];
+    for (let i = 0; i < printedGroup.length; i++) {
+      if (printedGroup[i + 1] && printedGroup[i + 1].needsParens) {
+        result.push(
+          "(",
+          printedGroup[i].printed,
+          printedGroup[++i].printed,
```

Comment:
Can you avoid doing the mutation here and instead do an explicit i++ and use i - 1 or something. Those kind of mutations are very error prone

Reply:
Done